### ISSUE TITLE
duckdns.org

### ISSUE THREAD

**Issue by bdr99** [easylist/easylist#14247]
### List the website(s) you're having issues:
All duckdns.org subdomains. I use duckdns.org to handle DNS for my personal instance of the home automation platform Home Assistant.

### What happens?
When I visit my Home Assistant instance on the duckdns.org subdomain I'm using, it fails to connect because its WebSocket connection is blocked by the below mentioned filter rule. If I host the same application on a domain that is not a subdomain of duckdns.org, it works fine.

Based on the comment in the file, it looks like this filter was added to block crypto miners, but blocking all websocket requests to all duckdns.org subdomains seems a bit extreme. Many people use duckdns to provide a DNS name for their self hosted applications, and any applications that use WebSockets won't work correctly with this filter. Therefore I think it should be removed.

### List Subscriptions you're using:
Default filters that come with uBlock Origin. In this case, the filter is coming from easyprivacy_trackingservers.txt.

### Your settings

- OS/version: Windows 11
- Browser/version: Firefox 107
- Adblock Extension/version: uBlock Origin 1.45.2

### Other details:
This is the filter that is blocking the WebSocket requests. It is from easyprivacy_trackingservers.txt.
`||duckdns.org^$third-party,websocket`

**Comment by ryanbr** [comment]
We don't recommend using PI hole Easyprivacy. 

But to address the issue, it came from https://github.com/easylist/easylist/commit/b5cfa5472973505ea92670a7fece103fd67f9ace where sites hosted on `duckdns.org`  were being used for coinmining, creating random duckdns subdomains then hosting them to mine CPU cycles. When you block one, another will just pop up. That said, its our first complaint since 2018..

```
ws://cfcs1.duckdns.org/*
wss://cfcs1.duckdns.org/*
wss://cfceu.duckdns.org/*
wss://webwidgetz.duckdns.org/*
...
```

**Comment by bdr99** [comment]
Thanks for the reply. I definitely understand how difficult of a situation this is, considering that blocking one subdomain will only temporarily address the issue. However, IMHO, I still think it is a bit heavy-handed to block websockets for all duckdns.org subdomains, since there are many legitimate users of this service.

Also, while I may be the first person to report this issue here, I'm definitely not the first to encounter it. Here are some examples of discussions where users were confused by their private self-hosted apps not working properly because of this rule:

 - https://community.home-assistant.io/t/ha-doing-wrong-dns-requests/231500
 - https://community.home-assistant.io/t/unable-to-connect-to-home-assistant-via-nginx-reverse-proxy/382937/28
 - https://community.home-assistant.io/t/unable-to-connect-to-home-assistant-via-nginx-reverse-proxy/382937/31
 - https://bugzilla.mozilla.org/show_bug.cgi?id=1437622
 - https://stackoverflow.com/a/72629097

**Comment by danger89** [comment]
People also can't use [my domain](https://explorer.melroy.org) since also here the websocket is blocked by easylist! While it's fully legit way of using DuckDNS in order to have my domain name always point to the correct public IP address.

Please, fix this!

**Comment by DandelionSprout** [comment]
In regards to uBO, my semi-qualified understanding is that something like `@@*$websocket,domain=melroy.org` could've been added to either EasyPrivacy or to "uBlock Filters - Unbreak".

In regards to Pi-hole, the Pi-hole devs have shown *zero* willingness to communicate with or align syntaxes with any other extensions or listmakers, leading to accidents like these. Such that I don't think anyone in the main uBO/AdGuard/ABP community would recommend Pi-hole itself in the slightest. Such is Pi-hole's hostility to listmakers that I can't even find info on which lists that are included in Pi-hole, but EasyPrivacy doesn't appear to be among them regardless.

**Comment by bdr99** [comment]
I'm not sure I understand why Pi-hole is relevant to this discussion. I'm using uBlock Origin and I believe @danger89 is as well. The rule that blocks websockets on duckdns.org subdomains is part of the default set of filter rules for uBlock Origin.

**Comment by bdr99** [comment]
> In regards to uBO, my semi-qualified understanding is that something like `@@*$websocket,domain=melroy.org` could've been added to either EasyPrivacy or to "uBlock Filters - Unbreak".

I don't think this is a workable solution. Duck DNS is a service that allows users to easily create domains on-the-fly for their personal/hobbyist projects. I don't think it's reasonable to require users to request an exception to be added to the rule list for their hobbyist projects, or even to expect users to know that this is what they would need to do in order to fix their problem.

**Comment by DandelionSprout** [comment]
Oh, right, now I see that Ryan must've misunderstood the OP with his *"We don't recommend using PI hole Easyprivacy."*, despite the OP clearly saying uBO instead of Pi-hole. My fault.

**Comment by danger89** [comment]
> I'm not sure I understand why Pi-hole is relevant to this discussion. I'm not using Pi-hole, I'm using uBlock Origin and I believe @danger89 is as well. The rule that blocks websockets on duckdns.org subdomains is part of the default set of filter rules for uBlock Origin, which is coming from one of the rule lists in this repo.

Indeed I'm using uBlock Origin. And almost all my visitors are also using ublock. Which broke my site since websockets are broken by easy list.

I have several domains, whitelisting all legit domains in the world that are trying to use websockets would be a nightmare.

**Comment by april83c** [comment]
also having problems due to this, this rule should be removed. it's way too broad

**Comment by melroy89** [comment]
> also having problems due to this, this rule should be removed. it's way too broad

Yea. And it is STILL NOT FIXED.. brrr
